### PR TITLE
WEA-8 Add tabs in home

### PR DESCRIPTION
**Pull request by ayush-shta**
Also populated data in the tab fragments

### CODE REVIEW

**Comment by subash-poudel** [comment]
Fix merge conflict first.

File: app/src/main/java/com/ayush/weatherapp/customViews/WrapContentHeightViewPager.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package com.ayush.weatherapp.customViews;
+
+import android.content.Context;
+import android.support.v4.view.ViewPager;
+import android.util.AttributeSet;
+import android.view.View;
+
+//Reference: https://stackoverflow.com/questions/8394681/android-i-am-unable-to-have-viewpager-wrap-content
+public class WrapContentHeightViewPager extends ViewPager {
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/customViews/WrapContentHeightViewPager.java]: Is this absolutely necessary?

File: app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java [hunk 2]
```diff
@@ -107,41 +128,29 @@ private void setNavigationView() {
...
 
   @Override public void setDailyForeCast(List<DailyForecast.DailyData> dailyForecastList) {
-    for (DailyForecast.DailyData dailyData : dailyForecastList) {
-      addDailyForecastView(dailyData);
-    }
 
+    tabPagerAdapter.setDailyForecastData(dailyForecastList);
+
+    //get forecast detail of today
     DailyForecast.DailyData forecastDetailToday = dailyForecastList.get(0);
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HomeActivity.java:150]: why taking only 0

File: app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java [hunk 3]
```diff
@@ -44,19 +49,53 @@ public void onResponse(@NonNull Call<Forecast> call, @NonNull Response<Forecast>
...
+        geocodingAPIInterface.getLocationDetails(requestString);
+
+    reverseGeoLocationCall.enqueue(new Callback<ReverseGeoLocation>() {
+      String localityAddress;
+
+      @Override
+      public void onResponse(Call<ReverseGeoLocation> call, Response<ReverseGeoLocation> response) {
+        ReverseGeoLocation reverseGeoLocation = response.body();
+        List<Address> addressList = reverseGeoLocation.getAddresses();
+        List<AddressComponents> addressComponentsList = addressList.get(0).getAddressComponents();
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java]: make sure that there is element 0 first

File: app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java [hunk 4]
```diff
@@ -44,19 +49,53 @@ public void onResponse(@NonNull Call<Forecast> call, @NonNull Response<Forecast>
...
+
+        localityAddress = addressComponentsList.get(LOCALITY_INDEX).getLongName();
+
+        view.setLocality(localityAddress);
+      }
+
+      @Override public void onFailure(Call<ReverseGeoLocation> call, Throwable t) {
+        Timber.e("Request fetch locality failed");
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java]: .

File: app/src/main/java/com/ayush/weatherapp/home/HourlyForecastFragment.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+  @Nullable @Override
+  public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
+      Bundle savedInstanceState) {
+    View view = inflater.inflate(R.layout.forecast_fragment, container, false);
+    ButterKnife.bind(this, view);
+    return view;
+  }
+
+  public void setData(List<HourlyForecast.HourlyData> hourlyForeCastList) {
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/HourlyForecastFragment.java:31]: A better way is to send only 6 data to this fragment.

File: app/src/main/java/com/ayush/weatherapp/home/TabPagerAdapter.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+public class TabPagerAdapter extends FragmentPagerAdapter {
+  private final static int NUMBER_OF_TABS = 2;
+  private String tabTitles[] = new String[] { "Daily", "Hourly" };
+  private DailyForecastFragment dailyForecastFragment;
+  private HourlyForecastFragment hourlyForecastFragment;
+
+  TabPagerAdapter(FragmentManager fm) {
+    super(fm);
+    dailyForecastFragment = new DailyForecastFragment();
```
Comment (subash-poudel) [inline app/src/main/java/com/ayush/weatherapp/home/TabPagerAdapter.java]: Fragment to be provided from elsewhere

File: app/src/main/java/com/ayush/weatherapp/customViews/WrapContentHeightViewPager.java (hunk 1 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/customViews/WrapContentHeightViewPager.java]: yes. to make the view pager wrap content or else it is always match parent by default.

File: app/src/main/java/com/ayush/weatherapp/home/HourlyForecastFragment.java (hunk 5 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/home/HourlyForecastFragment.java:31]: :ok:

File: app/src/main/java/com/ayush/weatherapp/home/TabPagerAdapter.java (hunk 6 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/home/TabPagerAdapter.java]: :ok:

File: app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java (hunk 3 above)
Comment (ayush-shta) [reply app/src/main/java/com/ayush/weatherapp/home/HomePresenterImpl.java]: :ok:

**Pull request merged**